torch/utils/cpp_extension.py

```diff
@@ -275,14 +322,21 @@ def unix_wrap_compile(obj, src, ext, cc_args, extra_postargs, pp_opts):
             try:
                 original_compiler = self.compiler.compiler_so
                 if _is_cuda_file(src):
-                    nvcc = _join_cuda_home('bin', 'nvcc')
+                    nvcc = (_join_rocm_home('bin', 'hipcc') if ROCM_HOME else _join_cuda_home('bin', 'nvcc'))
                     if not isinstance(nvcc, list):
                         nvcc = [nvcc]
                     self.compiler.set_executable('compiler_so', nvcc)
                     if isinstance(cflags, dict):
                         cflags = cflags['nvcc']
-                    cflags = COMMON_NVCC_FLAGS + ['--compiler-options',
+                    if ROCM_HOME:
```

Comment:
I don't know much about how pytorch implements ROCM, so apologies if this is a silly question: can we use both AMD and nvidia gpus on the same machine, or is it mutually exclusive?

Reply:
Currently we don't support AMD and NVidia in the same PyTorch.